Stream Rebranding

nick-funk opened the pull request:
Main branch for all rebranding of the client stream.

nick-funk committed:
[CORL-901] Rebrand comment actions and body (#2889)

* [ADR] Use Relay over Apollo (#2881)

* chore: ADR on Relay and Apollo

* chore: typo

* chore: add neg cons

* chore: add another pos

* chore: remove neg, because apollo needed an introspection query as a build step too..

* chore: fix number

* chore: deco

* chore: typo

* chore: clarification

Co-authored-by: Wyatt Johnson <wyattjoh@gmail.com>

* chore: bump 6.0.0 (#2891)

* fix: added more checks to migration (#2894)

* Update stream username to new branding variables

CORL-901

* Update badges/tags to use new v2 controls

CORL-901

* Update timestamps to use new branding stream side

CORL-901

* Update stream edit action to new branding

CORL-901

* Update comment html body to new branding

CORL-901

* Update comment action buttons to new branding

CORL-901

* Update responsive styles for action buttons

CORL-901

* Set comment link colors to new branding blue

CORL-901

* Add children prop on ButtonsBar to align with tests

CORL-901

* Make Reported button text bolded and vivid red

CORL-901

* Make reaction buttons stream blue when reacted

CORL-901

* Make user badges bold text

CORL-901

* Set margins on comment action buttons

CORL-901

* Set aria-labels on reaction buttons

Ensures that we always have a usable aria
even when the reaction button is just an icon.

CORL-901

* Set aria-label's on Reply and Share button's

CORL-901

* Update tests to match stream branding changes

CORL-901

* Update color of "Read More of this conversation"

CORL-901

Co-authored-by: Vinh <vinh@vinh.tech>
Co-authored-by: Wyatt Johnson <wyattjoh@gmail.com>

nick-funk committed:
[CORL-942] In-Stream Mod rebranding (#2902)

* [ADR] Use Relay over Apollo (#2881)

* chore: ADR on Relay and Apollo

* chore: typo

* chore: add neg cons

* chore: add another pos

* chore: remove neg, because apollo needed an introspection query as a build step too..

* chore: fix number

* chore: deco

* chore: typo

* chore: clarification

Co-authored-by: Wyatt Johnson <wyattjoh@gmail.com>

* chore: bump 6.0.0 (#2891)

* fix: added more checks to migration (#2894)

* Update stream username to new branding variables

CORL-901

* Update badges/tags to use new v2 controls

CORL-901

* Update timestamps to use new branding stream side

CORL-901

* Update stream edit action to new branding

CORL-901

* Update comment html body to new branding

CORL-901

* Update comment action buttons to new branding

CORL-901

* Update responsive styles for action buttons

CORL-901

* Set comment link colors to new branding blue

CORL-901

* Add children prop on ButtonsBar to align with tests

CORL-901

* Make Reported button text bolded and vivid red

CORL-901

* Make reaction buttons stream blue when reacted

CORL-901

* Make user badges bold text

CORL-901

* Set margins on comment action buttons

CORL-901

* Set aria-labels on reaction buttons

Ensures that we always have a usable aria
even when the reaction button is just an icon.

CORL-901

* Set aria-label's on Reply and Share button's

CORL-901

* Update tests to match stream branding changes

CORL-901

* Update color of "Read More of this conversation"

CORL-901

* Update caret container to new branding

CORL-942

* Update moderation menu buttons to new branding

CORL-942

* Update in-stream ban popover to new branding

CORL-942

* Update reject tombstone to new branding

CORL-942

Co-authored-by: Vinh <vinh@vinh.tech>
Co-authored-by: Wyatt Johnson <wyattjoh@gmail.com>

nick-funk committed:
[CORL-939] Rebrand RTE and reply (#2904)

* [ADR] Use Relay over Apollo (#2881)

* chore: ADR on Relay and Apollo

* chore: typo

* chore: add neg cons

* chore: add another pos

* chore: remove neg, because apollo needed an introspection query as a build step too..

* chore: fix number

* chore: deco

* chore: typo

* chore: clarification

Co-authored-by: Wyatt Johnson <wyattjoh@gmail.com>

* chore: bump 6.0.0 (#2891)

* fix: added more checks to migration (#2894)

* Update stream username to new branding variables

CORL-901

* Update badges/tags to use new v2 controls

CORL-901

* Update timestamps to use new branding stream side

CORL-901

* Update stream edit action to new branding

CORL-901

* Update comment html body to new branding

CORL-901

* Update comment action buttons to new branding

CORL-901

* Update responsive styles for action buttons

CORL-901

* Set comment link colors to new branding blue

CORL-901

* Add children prop on ButtonsBar to align with tests

CORL-901

* Make Reported button text bolded and vivid red

CORL-901

* Make reaction buttons stream blue when reacted

CORL-901

* Make user badges bold text

CORL-901

* Set margins on comment action buttons

CORL-901

* Set aria-labels on reaction buttons

Ensures that we always have a usable aria
even when the reaction button is just an icon.

CORL-901

* Set aria-label's on Reply and Share button's

CORL-901

* Update tests to match stream branding changes

CORL-901

* Update color of "Read More of this conversation"

CORL-901

* Update caret container to new branding

CORL-942

* Update moderation menu buttons to new branding

CORL-942

* Update in-stream ban popover to new branding

CORL-942

* Update reject tombstone to new branding

CORL-942

* Update replyTo and rte to new branding

CORL-939

Co-authored-by: Vinh <vinh@vinh.tech>
Co-authored-by: Wyatt Johnson <wyattjoh@gmail.com>

nick-funk committed:
Update message box to new branding

CORL-981

nick-funk committed:
Update user profile comment history to new branding

CORL-983

nick-funk committed:
Update stream tabs to new branding

CORL-982

nick-funk committed:
Update my profile tab bar to new branding

CORL-902

nick-funk committed:
Update download comments to new branding

CORL-902

nick-funk committed:
The Lord of the Buttons: Return of the Buttons

nick-funk committed:
Update the notification settings area to new branding

CORL-984

nick-funk committed:
Update the ignored users area to new branding

CORL-984

nick-funk committed:
Update change username to new branding

CORL-984

nick-funk committed:
Fixing snaps after rebase

nick-funk committed:
Use label instead of div for InputLabel's

Root element is now a label so that axe
will stop complaining that it can't find
the 'for' id on the label elements.

CORL-984

nick-funk committed:
More spacing tweaks for change username

CORL-903

nick-funk committed:
Update change email address to new branding

CORL-984

nick-funk committed:
Update change password to new branding

CORL-984

nick-funk committed:
Flesh out colour styles for v3 buttons

CORL-984

nick-funk committed:
Preliminarily update delete account to new branding

CORL-984

nick-funk committed:
Ensure client CSS style compatibility

CORL-984

nick-funk committed:
Update delete flow to new branding

CORL-984

nick-funk committed:
Add client classes to the delete account modal

CORL-984

nick-funk committed:
Further tweaks to the delete account modal

Modal callouts updated to new branding.
Deletion request callout updated to new
branding.
Fixing up tests that somehow were passing
before?
Fixing some margins and spacing around
deletion section.

nick-funk committed:
Move the account deletion callout to top

Puts it at the top of the my profile tab
so that it's clearly visible you are
deleting your account.

CORL-998

nick-funk committed:
Add first-class icon and title support to callouts

CORL-969

nick-funk committed:
Bold checkbox text when checked

CORL-984

nick-funk committed:
Finalize v3 buttons to match spec

This should be the last major change to
v3 buttons as the spec seems to have
solidified.

CORL-969

nick-funk committed:
Add in new stream red to colours

Use it for the negative v3 callouts

CORL-969

nick-funk committed:
Touch up some spacing on delete account

CORL-984

nick-funk committed:
Create tombstone component

Use it for ignored and deleted account
tombstones instead of the old callout.

CORL-969

nick-funk committed:
Use new tombstones in comment stream

CORL-969

nick-funk committed:
Update banned callout to new branding

CORL-969

nick-funk committed:
Set callout body text to regular font weight

CORL-969

nick-funk committed:
Update suspended callout to new branding

CORL-969

nick-funk committed:
Update closed stream callouts to new branding

CORL-969

nick-funk committed:
Update in-moderation callouts to new branding

CORL-969

nick-funk committed:
Update stream validation message to v3

CORL-969

nick-funk committed:
Improve ignored commenter management

Show "you are no longer ignoring" a
commenter when you choose to no longer
ignore them.

CORL-457

nick-funk committed:
Prettier cleanup after rebase off master

Somehow this was missed, probably needed to
kill and reload VSCode before I committed the
rebase.

nick-funk committed:
Switch previous callouts to use icon and title

CORL-969

nick-funk committed:
Update configure checkboxes to new branding

CORL-903

nick-funk committed:
Update Config > Q&A to new branding

CORL-903

nick-funk committed:
Update Config > Open/Close Stream to new branding

CORL-903

nick-funk committed:
Add validation message to config stream checkboxes

CORL-903

nick-funk committed:
Remove double list handling for Pref>Ignored

CORL-969

nick-funk committed:
Show experimental tag beside Q&A config

CORL-1051

nick-funk committed:
Move live update setting to its own section

CORL-903

nick-funk committed:
Remove moderation link from config area

CORL-903

nick-funk committed:
De-capitalize all the labels in configure

CORL-903

nick-funk committed:
Update message box config to new branding

Also moves it to new location in config.

CORL-903

nick-funk committed:
Re-brand the sign in/out field on the stream

CORL-1042

nick-funk committed:
Update tests for add message flow

CORl-903

nick-funk committed:
Rebrand Sign In/Up modals

CORL-999

nick-funk committed:
Re-brand SSO sign in/up modals

CORL-999

nick-funk committed:
Re-brand stream sorting controls

CORL-1043

nick-funk committed:
Create new stream focused colorFromMeta variant

nick-funk committed:
Rebrand go to moderation buttons

CORL-799

nick-funk committed:
Rebrand stream moderation drop-down

CORL-799

nick-funk committed:
Rebrand the report comment flow

Expands the report comment dialogs under
the comment instead of as a popover.
Also adds in the new branding styles.

CORL-941

nick-funk committed:
Rebrand the main stream tab bar

CORL-799

nick-funk committed:
Add new abusive reporting options

CORL-1009

nick-funk committed:
Rebrand featured comments

CORL-968

nick-funk committed:
Rebrand the community guidelines

CORL-1041

nick-funk committed:
Rebrand the share button

CORL-940

nick-funk committed:
Update style names on buttons

marginSize -> paddingSize
textSize -> fontSize

nick-funk committed:
Rebrand single conversation view

CORL-904

nick-funk committed:
Make indented comments grey coloured

CORL-904

nick-funk committed:
Make indented comment borders thinner

CORL-904

nick-funk committed:
Tweak colours/margins on single conversation view

CORL-904

nick-funk committed:
Add show comment on single conversation tombstones

CORL-412

nick-funk committed:
Rebrand single conversation view title

CORL-904

nick-funk committed:
Add clean target to package.json

Allows us to direct call gulp clean easily.

nick-funk committed:
Rebrand client side not found page

CORL-1025

nick-funk committed:
Rebrand password reset flow

CORL-1025

nick-funk committed:
Rebrand unsubscribe from notifications flow

CORL-1025

nick-funk committed:
Rebrand email confirmation flow

CORL-1025

nick-funk committed:
Rebrand download comments flow

CORL-1025

nick-funk committed:
Fix up account tests after rebrand

CORL-1025

nick-funk committed:
Add COMMENT_REPORTED_ABUSIVE to moderate card fragment

nick-funk committed:
Rebrand user ignore flow

Somehow this slipped through the cracks, fixed
it up now

tessalt commented on src/core/client/account/routes/NotFound.tsx:
```diff
@@ -1,12 +1,12 @@
-    <Typography variant="heading3" container="h1">
-      Not Found
-    </Typography>
+    <div className={styles.content}>Not Found</div>
```
out of scope but do we have a translation for this?

tessalt committed:
[CORL 1010] Collapse comment threads (#2976)

* toggle open and closed comment + replies

* show username and date for collapsed comment

* toggle comments and replies open and closed

* keep collapsed state of comments in reply list

* style collapsed comment toggle

* add css class overrides

* style answered comments when collapsed

* adjust spacing on collapse button:

* add hover styles

* add aria labels and update snapshots

tessalt commented on src/core/client/stream/App/TabBar.tsx:
```diff
@@ -22,45 +24,103 @@ const AppTabBar: FunctionComponent<Props> = (props) => {
+              <div>
+                <Icon
+                  size="lg"
+                  aria-label={
```
should this be translated?

tessalt commented on src/core/client/stream/App/TabBar.tsx:
```diff
@@ -22,45 +24,103 @@ const AppTabBar: FunctionComponent<Props> = (props) => {
           })}
           tabID="PROFILE"
+          variant="streamPrimary"
+          ariaLabel="My Profile"
```
translation key

tessalt commented on src/core/client/stream/App/TabBar.tsx:
```diff
@@ -22,45 +24,103 @@ const AppTabBar: FunctionComponent<Props> = (props) => {
+                  <Localized id="general-tabBar-configure">
+                    <div className={styles.smallText}>Configure</div>
+                  </Localized>
+                </div>
```
ensure aria-label properties are translated throughout this file

tessalt reviewed (changes requested):
Looks great!! 

- there are a few places with missing translation keys for aria-label values

- two visual issues I came across: 

<img width="385" alt="Screen Shot 2020-06-08 at 3 25 18 PM" src="https://user-images.githubusercontent.com/1789029/84072049-90598680-a99c-11ea-8e6c-9f00d59d6e8e.png">
edge of the "configure" tab on mobile

<img width="858" alt="Screen Shot 2020-06-08 at 3 24 32 PM" src="https://user-images.githubusercontent.com/1789029/84072067-98192b00-a99c-11ea-9d55-159dbbc5db34.png">

double divider on my comments tab

nick-funk committed:
Fix tests and snaps after rebase

nick-funk committed:
Make main stream tabs visible on mobile (iPhone 5)

nick-funk committed:
Remove double horizontal rule under My Comments

Horizontal rule wasn't conditionally showing
when we toggled ability to let users download
comments on and off.

nick-funk committed:
Add aria labelling to main stream tabs

Can now pass in a localization id to assign
aria-label/title to the tab.

nick-funk committed:
Add localization to account not found

nick-funk committed:
Update snapshots
